src/app/services/application.service.ts

```diff
@@ -33,6 +33,23 @@ export class ApplicationService {
       .pipe(catchError(this.errorHandler));
   }
 
+  authenticate(credentials, callback) {
+
+    const headers = new HttpHeaders(credentials ? {
+      authorization : 'Basic ' + btoa(credentials.username + ':' + credentials.password)
+    } : {});
+
+    this.httpClient.get('user', {headers}).subscribe(response => {
+      if (response['name']) {
```

Comment:
cia meta err del ['name'], siulo keist i response.name kurio neranda

Reply:
Add type to response - fix error.